### ISSUE TITLE
 Why the TensorRT mdoel doesn't show class names(model.names)? #4461 

### ISSUE THREAD

**Issue by miladfa7** [ultralytics/ultralytics#4467]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have converted the model to TensorRT format using this code

model.export(format='engine',  half=True, imgsz=[544,960], batch=10, device=0)

but after exporting model, I want to see class names but it shows just None value as result.

tensorrt_path ="train4/weights/best.engine"
trt_model = YOLO(tensorrt_path)
print(trt_model.names)

Output:
None

Also when the tensorrt model used for inference, the model has not detected any object. results model was empty.
I mean.
results = trt_model.predict(frames, imgsz=[544,960], batch=10, device=0)

**Comment by glenn-jocher** [comment]
@miladfa7 thank you for bringing this issue to our attention. It seems that there might be an issue with the way the TensorRT model handles class names. When a model is converted to TensorRT format, the class names are not carried over by default. 

To access class names, you would typically keep a separate file containing the class names (usually a '.names' file). This file is not encoded into the TensorRT model file, but should be kept alongside your model for reference when needed. You can load this file separately in your code to get the class names.

Regarding the empty results from your TensorRT model, it's possible that the issue may lie in your model's conversion or inferencing parameters. I would suggest you to re-visit the parameters and ensure that they match the expectations of the model.

Remember, precision while exporting to TensorRT model and the device used for inference should be the same. If you have trained the model with half-precision, try to ensure that your inference also uses the same.

Hope this helps! If you still face issues, please don't hesitate to reach out.

**Comment by i718108955** [comment]
> @miladfa7 thank you for bringing this issue to our attention. It seems that there might be an issue with the way the TensorRT model handles class names. When a model is converted to TensorRT format, the class names are not carried over by default.
> 
> To access class names, you would typically keep a separate file containing the class names (usually a '.names' file). This file is not encoded into the TensorRT model file, but should be kept alongside your model for reference when needed. You can load this file separately in your code to get the class names.
> 
> Regarding the empty results from your TensorRT model, it's possible that the issue may lie in your model's conversion or inferencing parameters. I would suggest you to re-visit the parameters and ensure that they match the expectations of the model.
> 
> Remember, precision while exporting to TensorRT model and the device used for inference should be the same. If you have trained the model with half-precision, try to ensure that your inference also uses the same.
> 
> Hope this helps! If you still face issues, please don't hesitate to reach out.

Hi there, I'm wondering how can I get or make the '.names' file? This is what I'm doing now:
for result in results:
            boxes = result.boxes
            for box in boxes:
                conf = box.conf
                if conf < confidence_threshold:
                    continue  # Skip this box if confidence is below the threshold
                b = box.xyxy[0]  # get box coordinates in (top, left, bottom, right) format
                c = box.cls
                if model.model.names[int(c)] == 'person':
                    x1, y1, x2, y2 = [int(x) for x in b]
                    object_depth = depth_image[int((y1 + y2) / 2), int((x1 + x2) / 2)]
                    object_distance = object_depth * 0.001  # to convert millimeters to meters
                    human_distances.append(object_distance)
                    cv2.rectangle(self.rgb_image, (x1, y1), (x2, y2), (255, 0, 0), 2)
                    cv2.putText(self.rgb_image, f'{model.names[int(c)]} {object_distance:.2f}m', (x1, y1 - 10),
                                cv2.FONT_HERSHEY_SIMPLEX, 0.5, (255, 0, 0), 2)

when I use the '.pt' file, it works good, but I can't find the 'person' tag on '.engine' file, I know there is must be the 'name' tags in my '.engine' file, because I try 'Multi-Object Tracking with Ultralytics YOLO' with my '.engine' file it  will show the tags like 'Car 93%' upon the boxes, so how can I make the '.names' file work with my code?

**Comment by Burhan-Q** [comment]
This issue was addressed in #9439 and you can now access the class names with an `engine` model.

```py
from ultralytics import YOLO

model = YOLO("yolov8n.engine")
mode.names

>>> Loading yolov8n.engine for TensorRT inference...
>>> [05/07/2024-12:48:57] [TRT] [I] Loaded engine size: 7 MiB
>>> [05/07/2024-12:48:57] [TRT] [I] [MemUsageChange] TensorRT-managed allocation in IExecutionContext creation: CPU +0, GPU +158, now: CPU 0, GPU 161 (MiB)
>>> {0: 'person', 1: 'bicycle', 2: 'car', 3: 'motorcycle', 4: 'airplane', 5: 'bus', 6: 'train', 7: 'truck', 8: 'boat', 9: 'traffic light', 10: 'fire hydrant', 11: 'stop sign', 12: 'parking meter', 13: 'bench', 14: 'bird', 15: 'cat', 16: 'dog', 17: 'horse', 18: 'sheep', 19: 'cow', 20: 'elephant', 21: 'bear', 22: 'zebra', 23: 'giraffe', 24: 'backpack', 25: 'umbrella', 26: 'handbag', 27: 'tie', 28: 'suitcase', 29: 'frisbee', 30: 'skis', 31: 'snowboard', 32: 'sports ball', 33: 'kite', 34: 'baseball bat', 35: 'baseball glove', 36: 'skateboard', 37: 'surfboard', 38: 'tennis racket', 39: 'bottle', 40: 'wine glass', 41: 'cup', 42: 'fork', 43: 'knife', 44: 'spoon', 45: 'bowl', 46: 'banana', 47: 'apple', 48: 'sandwich', 49: 'orange', 50: 'broccoli', 51: 'carrot', 52: 'hot dog', 53: 'pizza', 54: 'donut', 55: 'cake', 56: 'chair', 57: 'couch', 58: 'potted plant', 59: 'bed', 60: 'dining table', 61: 'toilet', 62: 'tv', 63: 'laptop', 64: 'mouse', 65: 'remote', 66: 'keyboard', 67: 'cell phone', 68: 'microwave', 69: 'oven', 70: 'toaster', 71: 'sink', 72: 'refrigerator', 73: 'book', 74: 'clock', 75: 'vase', 76: 'scissors', 77: 'teddy bear', 78: 'hair drier', 79: 'toothbrush'}
